pkg/cmd/taskrun/cancel_test.go

```diff
@@ -0,0 +1,107 @@
+package taskrun
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/tektoncd/cli/pkg/test"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1"
+	"github.com/tektoncd/pipeline/pkg/reconciler/v1alpha1/pipelinerun/resources"
+	pipelinetest "github.com/tektoncd/pipeline/test"
+	tb "github.com/tektoncd/pipeline/test/builder"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	k8stest "k8s.io/client-go/testing"
+	"knative.dev/pkg/apis"
+)
+
+func TestTaskRunCancel(t *testing.T) {
+	seeds := make([]pipelinetest.Clients, 0)
+	failures := make([]pipelinetest.Clients, 0)
+
+	for i := 0; i < 1; i++ {
+		trs := []*v1alpha1.TaskRun{
+			tb.TaskRun("taskrun-1", "ns",
+				tb.TaskRunLabel("tekton.dev/task", "task"),
+				tb.TaskRunSpec(tb.TaskRunTaskRef("task")),
+				tb.TaskRunStatus(
+					tb.StatusCondition(apis.Condition{
+						Status: corev1.ConditionTrue,
+						Reason: resources.ReasonSucceeded,
+					}),
+				),
+			),
+		}
+		trs2 := []*v1alpha1.TaskRun{
+			tb.TaskRun("failure-taskrun-1", "ns",
+				tb.TaskRunLabel("tekton.dev/task", "failure-task"),
+				tb.TaskRunSpec(tb.TaskRunTaskRef("failure-task")),
+				tb.TaskRunStatus(
+					tb.StatusCondition(apis.Condition{
+						Status: corev1.ConditionTrue,
+						Reason: resources.ReasonSucceeded,
+					}),
+				),
+			),
+		}
+		cs, _ := test.SeedTestData(t, pipelinetest.Data{TaskRuns: trs})
+		cs2, _ := test.SeedTestData(t, pipelinetest.Data{TaskRuns: trs2})
+
+		cs2.Pipeline.PrependReactor("update", "taskruns", func(action k8stest.Action) (bool, runtime.Object, error) {
```

Comment:
Wonders why we need this? :thinking:

Reply:
Thank you for your review!!
It is for making update error and just improve coverage.
It's test following code.
```
tr.Spec.Status = v1alpha1.TaskRunSpecStatusCancelled
if _, err := cs.Tekton.TektonV1alpha1().TaskRuns(p.Namespace()).Update(tr); err != nil {
	return fmt.Errorf("failed to cancel taskrun %q: %s", trName, err)
}
```

Reply:
:+1: got it thanks!

Gave me the impression that every time update call would lead to an error state?
How about? 

```
testParams := []struct {
		name      string
		command   []string
		input     []v1alpha1.TaskRun
		wantError bool
		want      string
	}{
		{
			name:      "Canceling taskrun successfully",
			command:   []string{"cancel", "taskrun-1", "-n", "ns"},
			input:     seeds[0],
			wantError: false,
			want:      "TaskRun cancelled: taskrun-1\n",
		},
                ....
	}

       for _, t := range testParams {
                cs := seedData()
                if wantErr {
                       prependReactor()
                }
       }

```

Reply:
PrependReactor is called on only cs2, and cs2 is added to failures slices.
successful case is using seeds slice as a input, and failure case is using failure slice as a input.
so update call would be error only failure test cases :thumbsup: !

```
seeds = append(seeds, cs)
	failures = append(failures, cs2)

	testParams := []struct {
		name      string
		command   []string
		input     pipelinetest.Clients
		wantError bool
		want      string
	}{
		{
			name:      "Canceling taskrun successfully",
			command:   []string{"cancel", "taskrun-1", "-n", "ns"},
			input:     seeds[0],
			wantError: false,
			want:      "TaskRun cancelled: taskrun-1\n",
		},
		{
			name:      "Not found taskrun",
			command:   []string{"cancel", "nonexistent", "-n", "ns"},
			input:     seeds[0],
			wantError: true,
			want:      "failed to find taskrun: nonexistent",
		},
		{
			name:      "Failed canceling taskrun",
			command:   []string{"cancel", "failure-taskrun-1", "-n", "ns"},
			input:     failures[0],
			wantError: true,
			want:      "failed to cancel taskrun \"failure-taskrun-1\": test error",
		},
```

Reply:
cool no worries :+1:

Reply:
Thanks !!